js/builder_graph.js

```diff
@@ -445,20 +445,34 @@ class PowderInputNode extends InputNode {
 
     compute_func(input_map) {
         // TODO: haha improve efficiency to O(n) dumb
-        // also, error handling is missing
         let input = this.input_field.value.trim();
         let powdering = [];
         let errorederrors = [];
         while (input) {
             let first = input.slice(0, 2);
             let powder = powderIDs.get(first);
             if (powder === undefined) {
-                return null;
+                if (first.length > 0)
+                    errorederrors.push(first);
+                else
+                    break;
             } else {
                 powdering.push(powder);
             }
             input = input.slice(2);
         }
+
+        if (this.input_field.getAttribute("placeholder") != null) {
+            let placeholder = this.input_field.getAttribute("placeholder");
+            if (placeholder.includes(" slots") && parseInt(placeholder[0]) < powdering.length)
+                errorederrors.push("Too many powders: " + powdering.length);
+        }
+
+        if (errorederrors.length)
+            this.input_field.classList.add("is-invalid");
+        else
+            this.input_field.classList.remove("is-invalid");
+
```

Comment:
should this still return a valid powdering if the input errored out?

Reply:
I changed it so that it returns null if there is an error, but this created an error about not loading everything when I viewed the website.
Also, if you get one powder wrong, it still computes the other powders, which could be useful.

Reply:
I see. That makes sense

maybe I can look into the load issue later -- it should still load, hmm.